src/Viewer.tsx

```diff
@@ -317,13 +320,75 @@ function Viewer(): ReactElement {
     [featureThresholds, config.keepRangeBetweenDatasets]
   );
 
+  // DATASET LOADING ///////////////////////////////////////////////////////
+  /**
+   * Replaces the current dataset with another loaded dataset. Handles cleanup and state changes.
+   * @param newDataset the new Dataset to replace the existing with. If null, does nothing.
+   * @param newDatasetKey the key of the new dataset in the Collection.
+   * @returns a Promise<void> that resolves when the loading is complete.
+   */
+  const replaceDataset = useCallback(
+    async (newDataset: Dataset | null, newDatasetKey: string): Promise<void> => {
+      console.trace("Replacing dataset with " + newDatasetKey + ".");
+      // TODO: Change the way flags are handled to prevent flickering during dataset replacement
+      setDatasetOpen(false);
+      if (newDataset === null) {
+        // TODO: Determine with UX what expected behavior should be for bad datasets
+        return;
+      }
+
+      // Dispose of the old dataset
+      if (dataset !== null) {
+        dataset.dispose();
+      }
+      // State updates
+      setDataset(newDataset);
+      setDatasetKey(newDatasetKey);
+
+      // Only change the feature if there's no equivalent in the new dataset
+      let newFeatureName = featureName;
+      if (!newDataset.hasFeature(newFeatureName)) {
+        newFeatureName = newDataset.featureNames[0];
+      }
+      replaceFeature(newDataset, newFeatureName);
+      resetColorRampRangeToDefaults(newDataset, newFeatureName);
+      setFeatureName(newFeatureName);
+
+      await canv.setDataset(newDataset);
+      canv.setFeature(newFeatureName);
+
+      // Clamp frame to new range
+      const newFrame = Math.min(currentFrame, canv.getTotalFrames() - 1);
+      await setFrame(newFrame);
+
+      setFindTrackInput("");
+      if (selectedBackdropKey && !newDataset.hasBackdrop(selectedBackdropKey)) {
+        setSelectedBackdropKey(null);
+      }
+      setSelectedTrack(null);
+      setDatasetOpen(true);
+      setFeatureThresholds(validateThresholds(newDataset, featureThresholds));
+      console.log("Num Items:" + newDataset?.numObjects);
```

Comment:
Fixed a bug on this line, used to be `dataset?.numObjects` which would try and get the number of objects from a non-existent dataset.

Reply:
Is this log statement intended for debugging or monitoring purposes?